keps/sig-node/753-sidecar-containers/README.md

```diff
@@ -1080,8 +1244,38 @@ in back-to-back releases.
 - Deprecate the flag
 -->
 
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+- E2e testing of the existing scenarios with and without the feature gate turned
+  on
+
+#### Beta
+
+- Implement proper termination ordering.
+- Provide defaults for `restartPolicy` field on init containers, `nil` is not
```

Comment:
If that's true, why did we use a pointer?

Reply:
this has to merge for beta:
https://github.com/kubernetes/kubernetes/pull/120176

Reply:
I think the PR is not relevant to this.

We may have something like `inheritPodRestartPolicy`by default?
I'd say nil is OK to indicate it is inherited from the pod's restartPolicy.

Reply:
If we do that, initContainers will become sidecars for all Pods created by ReplicaSets.
We should default to `Never` WDYT?

Reply:
It depends on the pod's restartPolicy.

If the pod has restartPolicy of 'Never', its regular init containers will never restart.

If the pod has restartPolicy of 'OnFailure', or 'Always', its regular init containers will restart only on failure.

Not sure how to name it. How about just call it 'Default' and document the detail...? (just like the 'Default' of PodDNSConfig)

@SergeyKanzhelev WDYT?

Reply:
I see, so it basically means "I'm not a sidecar". Hmm, I would still prefer `nil` then...